### DISCUSSION TITLE
so-rule-update with self signed certificates

### DISCUSSION THREAD

**Discussion by Jackson-Pollock** [Security-Onion-Solutions/securityonion#6869] [Unsupported Versions] [answered]
Hi,

I trusted my self-signed cert in to /etc/pki/ca-trust/source/anchors/ but still getting `certificate verify failed: self signed certificate in certificate chain (_ssl.c:1129)` error

**Comment by Jackson-Pollock** [comment]
I tried this https://github.com/Security-Onion-Solutions/securityonion/discussions/6364
still same certificate error. I've proxy behind SO and added proxy cert in /etc/pki/ca-trust/source/anchors/ I am able to `wget https://rules.emergingthreats.net/open/suricata-6.0.0/emerging.rules.tar.gz.` but not able to `so-rule-update`

**Comment by defensivedepth** [comment, marked as answer]
Did you setup proxy config through the Security Onion setup?

**Reply by Jackson-Pollock** [reply]
No, That's Zscaler proxy is running on endpoint and SO VM is running on top of it.

**Reply by Jackson-Pollock** [reply]
How to give certificate path to idstool docker so that it can read that proxy certificate? like how we do `curl` giving `--cacert` parameter. 
I tried this command `docker exec so-idstools /bin/bash -c "CURL_CA_BUNDLE=''; cd /opt/so/idstools/etc && idstools-rulecat --force"` but same error `SSL: CERTIFICATE_VERIFY_FAILED`

**Reply by defensivedepth** [reply]
To be clear,  idstools does not use curl to pull the rules. It's a python app and it's using the built in library urllib.

**Reply by rwwiv** [reply]
To add to what @defensivedepth is saying, the documentation for `idstools-rulecat` ([which can be found here](https://idstools.readthedocs.io/en/latest/tools/rulecat.html)) doesn't mention the ability to define the ca cert. [This is the relevant code](https://github.com/jasonish/py-idstools/blob/master/idstools/net.py#L40-L73) in idstools that actually pulls from a URL - in this case, the ET URL.

**Reply by Jackson-Pollock** [reply]
Ok understood! But how only idstool is producing this error and not wget/curl (I used wget just to verify if I can download rule file from emergingthreat without certificate error) I am really frustrated with this `SSL: CERTIFICATE_VERIFY_FAILED` error.

**Reply by defensivedepth** [reply]
Can you get OS package updates and other things just fine?

**Reply by Jackson-Pollock** [reply]
as an alternative worked with local url (`--url`)
